### PR TITLE
 #168781801 docker setup

### PR DESCRIPTION
**Pull request by alexismajya**
#### What does this PR do?
 Setting up the docker
#### Description of Task to be completed?
- Setting up `dockerfile`
- Setting up `docker-compose`
-Setting up `docker-compose.dev` for development
-Setting up `docker-compose.prod` for  production
- creating a `dockerignore`
- Add docker setup steps to the readme file

#### How should this be manually tested?
- Install docker on a local machine
 - run ` npm run dockerDev` for development
 - run ` npm run dockerProd` for production

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[Pivotal Tracker](https://www.pivotaltracker.com/story/show/168781801)
#### Questions:

### CODE REVIEW

**Review by Crispy-rw — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@alexismajya good job so far however you have not catered for production. Your docker commands are only for development kindly make research on how you can cater for both development and production.

File: dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ROM node:10.16.3
```
Comment (lemoissonM) [inline dockerfile]: try to correct this, It's FROM and not ROM

**Review by lemoissonM — CHANGES_REQUESTED** [review]

**Review by swaibat — APPROVED** [review]

File: dockerfile (hunk 1 above)
Comment (alexismajya) [reply dockerfile]: I fixed it

**Review by urbanishimwe — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   "scripts": {
     "start": "babel-node ./src/index.js",
     "dev": "nodemon --exec babel-node ./src/index",
+    "prod": "nodemon --exec babel-node ./src/index",
```
Comment (nakiwuge) [inline package.json]: This is not necessary, you can use the `start script` because we do not need nodemon in production.

Comment (alexismajya) [reply package.json]: Yes, we have said that we will change it after. but I will change it

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+version: "3"
+services:
+  app:
+    build: .
+    depends_on:
+      - postgres
+    environment:
+      NODE_ENV: development
+      DATABASE-URL: ${TEST-DB}
```
Comment (urbanishimwe) [inline docker-compose.dev.yml]: nice work. howerver, it is `TEST_DB` not `TEST-DB` please check the env.sample, also check on `DATABASE_URL`

**Review by urbanishimwe — CHANGES_REQUESTED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**